switch/ogre3d/ogre-switch.patch

```diff
@@ -0,0 +1,690 @@
++#include "OgreViewport.h"
++
++#include <switch.h>
++
++namespace Ogre {
++    SwitchEGLWindow::SwitchEGLWindow(SwitchEGLSupport *glsupport)
++        : EGLWindow(glsupport)
++    {
++        mClosed = true;
++    }
++
++    void SwitchEGLWindow::getLeftAndTopFromNativeWindow( int & left, int & top, uint width, uint height )
++    {
++        // We are always fixed at (0, 0) for the Nintendo Switch.
++        left = top = 0;
++    }
++
++    void SwitchEGLWindow::initNativeCreatedWindow(const NameValuePairList *miscParams)
++    {
++    }
++
++    void SwitchEGLWindow::createNativeWindow( int &left, int &top, uint &width, uint &height, String &title )
++    {
++    }
++
++    void SwitchEGLWindow::reposition( int left, int top )
++    {
++    }
++
++    void SwitchEGLWindow::resize(uint width, uint height)
++    {
++    }
++
++    void SwitchEGLWindow::windowMovedOrResized()
++    {
++    }
++
++    void SwitchEGLWindow::switchFullScreen(bool fullscreen)
++    {
++
++    }
++
++    void SwitchEGLWindow::create(const String& name, uint width, uint height,
++                               bool fullScreen, const NameValuePairList *miscParams)
++    {
++        // Most of these are fixed for the Nintendo Switch.
++        mName = name;
++        mLeft = 0;
++        mTop = 0;
++        mIsFullScreen = true;
++        mIsExternal = false;
++        mHwGamma = false;
++        mWidth = 1920;
++        mHeight = 1080;
++
++        // Create the window and GL context.
++        _notifySurfaceDestroyed();
++        _notifySurfaceCreated(nwindowGetDefault(), 0);
++    }
++
++    void SwitchEGLWindow::_notifySurfaceDestroyed()
++    {
++        if(mClosed)
++            return;
++
++        mContext->setCurrent();
++        mContext->_destroyInternalResources();
++
++        eglDestroySurface(mEglDisplay, mEglSurface);
++        EGL_CHECK_ERROR
++        mEglSurface = 0;
++
++        eglTerminate(mEglDisplay);
++        EGL_CHECK_ERROR
++        mEglDisplay = 0;
++
++        mActive = false;
++        mVisible = false;
++        mClosed = true;
++    }
++
++    void SwitchEGLWindow::_notifySurfaceCreated(void* window, void* config)
++    {
++        // Set the new window.
++        mWindow = window;
++
++        nwindowSetDimensions((NWindow*)mWindow, mWidth, mHeight);
++
++        mEglDisplay = mGLSupport->getGLDisplay();
++
++        EGLint numConfigs;
++        static const EGLint framebufferAttributeList[] =
++        {
++            EGL_RENDERABLE_TYPE, EGL_OPENGL_BIT,
++            EGL_RED_SIZE,     8,
++            EGL_GREEN_SIZE,   8,
++            EGL_BLUE_SIZE,    8,
++            EGL_ALPHA_SIZE,   8,
++            EGL_DEPTH_SIZE,   24,
++            EGL_STENCIL_SIZE, 8,
++            EGL_NONE
++        };
++
++        eglChooseConfig(mEglDisplay, framebufferAttributeList, &mEglConfig,
++            1, &numConfigs);
++
++        mEglSurface = eglCreateWindowSurface(mEglDisplay, mEglConfig,
++            (NWindow*)mWindow, nullptr);
```

Comment:
This `(NWindow*)` cast is superfluous because switch-mesa's EGL headers define the native window type to be `void*` as well.